Application/EdFi.Ods.Api/Filters/EnforceAssignedProfileUsageFilter.cs

```diff
@@ -160,10 +153,31 @@ void SetForbiddenResponse()
                 SecurityAuthorizationException.DefaultDetail,
                 errorMessage)
             {
-                CorrelationId = (string) _logContextAccessor.GetValue(CorrelationConstants.LogContextKey)
+                CorrelationId = _logContextAccessor.GetCorrelationId()
             }.AsSerializableModel();
 
             context.Result = new ObjectResult(problemDetails) { StatusCode = problemDetails.Status };
         }
+
+        string[] GetAssignedProfilesForRequest()
+        {
+            try
+            {
+                return apiClientContext.Profiles.Where(
+                        p => _profileResourceModelProvider.GetProfileResourceModel(p)
+                                .ResourceByName.TryGetValue(resourceFullName, out var contentTypes)
+                            && (relevantContentTypeUsage == ContentTypeUsage.Readable
+                                ? contentTypes.Readable
+                                : contentTypes.Writable)
+                            != null)
+                    .ToArray();
+            }
+            catch (FeatureDisabledException ex)
+            {
+                // Force a client configuration error response if profile(s) are assigned and the Profiles feature is disabled.
+                throw new ClientConfigurationException(ClientConfigurationException.DefaultDetail,
+                    $"The API client has been configured with an assigned profile but the '{ex.FeatureName}' feature is disabled.");
```

Comment:
This exception will never happen (as of now), right? Because `GetAssignedProfilesForRequest()` doesn't get called if the Profiles feature is disabled [because of this condition](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/bbd005dc9cb2a85e9b3928c2fd13c3a4b14c140f/Application/EdFi.Ods.Api/Filters/EnforceAssignedProfileUsageFilter.cs#L55).

Also note this is the only place where `ClientConfigurationException` is used.

Reply:
I have gone a bit further and done some refactoring around disabled feature handling generally, but for this particular change I am removing the unused exception class and the surrounding `try/catch` block. Seems like it could potentially be useful, but if that happens we can add it (or something similar) back.